guardian/commercial-templates
# Migrate fabric-custom-xl with common fabric component

## What does this change?
This PR serves as part of the larger body of work to migrate commercial templates from Legacy to Svelte.

Added:
-  New `svelte` ready template 
- Extracted the common HTML structure and styles from the current fabric-custom template and created a FabricCustom component.
Additionally:
- Removed `ThirdPartyTags` (was used for pulling a creative from a tag, which we don't use)

Addition information:
- Introduced an index.ts file alongside index.svelte in the fabric-custom-xl directory to handle server-side logic, ensuring compatibility with Node.js during server-side rendering (SSR).
- The index.svelte file focuses on client-side rendering, making the code more modular and maintainable by separating server-side tasks from client-side presentation.
This approach ensures that server-side logic is executed in the correct environment while keeping the client-side rendering logic clean and focused.



## How to test
Run locally and check the fabric-custom component renders as previously without any UI or behaviour difference. 



## How can we measure success?
Successfully render a template, identical to the legacy version in both UI and functionality.




## Images
Also tested in GAM with `?adtest=custom-xl-migration-test`
<img width="500" alt="Screenshot 2025-01-07 at 13 01 34" src="https://github.com/user-attachments/assets/319632bf-edce-4c43-8ed1-0c0f4a79eb47" />

| Before| After |
|---------|---------|
| <img width="513" alt="Screenshot 2024-12-11 at 12 43 13" src="https://github.com/user-attachments/assets/45694eaf-1576-4c60-be0a-57900364ec9c">| <img width="508" alt="Screenshot 2024-12-11 at 12 49 40" src="https://github.com/user-attachments/assets/947396dc-66a1-48d6-a5e8-d61ed7b96ded"> |


https://github.com/user-attachments/assets/3d8a1c91-c5b2-4b4a-b91b-67e4ffc85797




## Accessibility
